karabo_bridge/tests/test_client.py

```diff
@@ -48,3 +48,11 @@ def test_iterator(sim_server):
     for i, (data, metadata) in enumerate(islice(c, 3)):
         trainId = metadata['SPB_DET_AGIPD1M-1/DET/0CH0:xtdf']['timestamp.tid']
         assert trainId == 10000000000 + i
+
+
+def test_timeout():
+    no_server = 'ipc://nodata'
+    with Client(no_server, timeout=200) as c:
+        with pytest.raises(TimeoutError) as info:
+            tid, data = c.next()
+    assert 'No data received from ipc://nodata in the last 200 ms' in str(info.value)
```

Comment:
A more complete test would be to test that the client can send 'next' to request the data after a timeout request. In principle, it should go into the integration test if there is one. But for the unitest here, it could be improved by testing sending the `next` request more than once.

Reply:
Thanks for the suggestion, I improved the test.